src/main/java/io/debezium/connector/informix/InformixValueConverters.java

```diff
@@ -41,20 +51,106 @@ public InformixValueConverters(DecimalMode decimalMode, TemporalPrecisionMode te
+
+            if (column.scale().get() == 255 && decimalMode == DecimalMode.PRECISE) {
+                return VariableScaleDecimal.builder();
+            }
+
+            return super.schemaBuilder(column);
         }
+
+        if (decimalMode == DecimalMode.PRECISE) {
+            return VariableScaleDecimal.builder();
+        }
+
+        if (column.length() == 0) {
+            // Defined as DECIMAL without specifying a length and scale, treat as DECIMAL(16)
+            return SpecialValueDecimal.builder(decimalMode, 16, -1);
+        }
+
+        return SpecialValueDecimal.builder(decimalMode, column.length(), -1);
     }
 
     @Override
     public ValueConverter converter(Column column, Field fieldDefn) {
         switch (column.jdbcType()) {
+            case Types.NUMERIC:
+            case Types.DECIMAL:
+                return getNumericConverter(column, fieldDefn);
             default:
                 return super.converter(column, fieldDefn);
         }
     }
 
+    private ValueConverter getNumericConverter(Column column, Field fieldDefn) {
+        if (column.scale().isPresent()) {
+
+            if (column.scale().get() == 255 && decimalMode == DecimalMode.PRECISE) {
+                return data -> convertVariableScale(column, fieldDefn, data);
+            }
+
+            return data -> convertNumeric(column, fieldDefn, data);
+        }
+
+        return data -> convertVariableScale(column, fieldDefn, data);
+    }
+
+    private Object convertVariableScale(Column column, Field fieldDefn, Object data) {
+        data = convertNumeric(column, fieldDefn, data); // provides default value
+
+        if (data == null) {
+            return null;
+        }
+        if (decimalMode == DecimalMode.PRECISE) {
+            if (data instanceof SpecialValueDecimal) {
+                return VariableScaleDecimal.fromLogical(fieldDefn.schema(), (SpecialValueDecimal) data);
+            }
+            else if (data instanceof BigDecimal) {
+                return VariableScaleDecimal.fromLogical(fieldDefn.schema(), (BigDecimal) data);
+            }
+        }
+        else {
+            return data;
+        }
+        return handleUnknownData(column, fieldDefn, data);
+    }
+
+    @Override
+    protected Object convertString(Column column, Field fieldDefn, Object data) {
+        if (data instanceof Clob) {
+            return convertValue(column, fieldDefn, data, "", (receiver) -> {
+                try {
+                    receiver.deliver(new BufferedReader(((IfxCblob) data).getCharacterStream()).lines().collect(Collectors.joining(System.lineSeparator())));
+                }
+                catch (SQLException e) {
+                    throw new RuntimeException("Error processing data from " + column.jdbcType() + " and column " + column +
+                            ": class=" + data.getClass(), e);
+                }
+            });
+        }
+        return super.convertString(column, fieldDefn, data);
+    }
```

Comment:
I'm assuming this change is specifically for tests?  
It just caught my eye as it seems unrelated to `VariableScaleDecimal` behavior, no?

Reply:
Yes, I adapted the AbstractDatatypesTest from the Oracle-connector, and the CLOB needed some care